### DISCUSSION TITLE
How to reference Windows.Forms on Windows

### DISCUSSION THREAD

**Discussion by nielsbosma** [AvaloniaUI/Avalonia#7306] [Q&A] [answered]
I want to use the FolderBrowserDialog from System.Windows.Forms (much more user friendly than one used in Avalonia.Controls.OpenFolderDialog)

![image](https://user-images.githubusercontent.com/4034492/148055479-06c9a5a8-7cdb-41ee-a97e-ae3a7a05ffda.png)

How can I reference Windows.Forms? or is there a better way?

**Comment by kekekeks** [comment, marked as answer]
https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props-desktop#net-5-and-later-versions

**Comment by robloo** [comment]
Obviously do what is best for you. That said, this type of dialog is very, very old -- it's from the 90s. Many users don't expect that these days. It might be better to use the native file selection dialogs (instead of Avalonia's managed version).
